oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/VersionGarbageCollector.java

```diff
@@ -1222,6 +1229,29 @@ public void removeGarbage(final VersionGCStats stats) {
 
             timer.reset().start();
             try {
+                if (embeddedVerification) {
+                    // embedded verification is done completely independent of DocumentStore
+                    // also, it's done irrespective of dry-run or not
+                    final Iterator<UpdateOp> it = updateOpList.iterator();
+                    while(it.hasNext()) {
+                        final UpdateOp update = it.next();
+                        NodeDocument oldDoc = ds.find(Collection.NODES, update.getId());
```

Comment:
I think we should fetch them all in one go. That would be much faster.
Since no such API exists in `DocumentStore`, I suggest we use Mongo APIs directly here.

Reply:
agree that a bulk `find` would be handy. In this case it would mostly be reading from the `nodesCache` though, as detailedGC just needed the oldDoc as well. So it's mostly not going to read from mongo. 
What about splitting bulk API into a separate ticket if we see it is necessary?

Reply:
(I'd probably avoid creating a jira ticket already now though, only once we find out it is a performance problem)